internal/services/oracle/oracle_exadata_infrastructure_resource.go

```diff
@@ -324,6 +347,8 @@ func (ExadataInfraResource) Read() sdk.ResourceFunc {
 					state.StorageCount = pointer.From(props.StorageCount)
 					state.Shape = props.Shape
 					state.MaintenanceWindow = FlattenMaintenanceWindow(props.MaintenanceWindow)
+					state.DatabaseServerType = pointer.ToString(props.DatabaseServerType)
+					state.StorageServerType = pointer.ToString(props.StorageServerType)
```

Comment:
These are planned to be deprecated, so we use the generic replacement `From()` to extract a value from a pointer.
```suggestion
					state.DatabaseServerType = pointer.From(props.DatabaseServerType)
					state.StorageServerType = pointer.From(props.StorageServerType)
```